cpp/src/io/csv/reader_impl.cu

```diff
@@ -120,62 +120,57 @@ string removeQuotes(string str, char quotechar)
 }
 
 /**
- * @brief Parse the first row to set the column names in the raw_csv parameter.
- * The first row can be either the header row, or the first data row
+ * @brief Parse a row of input to get the column names. The row can either be the header, or the
+ * first data row. If the header is not used, column names are generated automatically.
  */
-std::vector<std::string> get_column_names(std::vector<char> const& header,
+std::vector<std::string> get_column_names(std::vector<char> const& row,
                                           parse_options_view const& parse_opts,
                                           int header_row,
                                           std::string prefix)
 {
   std::vector<std::string> col_names;
 
-  // If there is only a single character then it would be the terminator
-  if (header.size() <= 1) { return col_names; }
-
-  std::vector<char> first_row = header;
+  // Empty row, return empty column names vector
+  if (row.empty() or (row.size() == 1 and row[0] == '\n')) { return col_names; }
```

Comment:
Should this check for an empty row as `\r\n` as well?

Reply:
Similar question - should we consider the line terminator character from the reader options as well if it is different from `\n, \r, \r\n`?

Reply:
Would just a carriage return also be sufficient?

Reply:
Starting to look like this line has more bugs than the Amazon rainforest.
I'll bring it back to the lab and test these cases.